src/utils/ota/restart_windows.go

```diff
@@ -0,0 +1,46 @@
+//go:build windows
+// +build windows
+
+package ota
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+func Restart(extraArgs ...string) error {
+	execPath, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("failed to locate the executable file: %w", err)
+	}
+
+	// A specific way the `cmd.exe` processes the `start` command: it takes the
+	// first quoted substring as a process name! (https://superuser.com/a/1656458)
+	cmdLine := fmt.Sprintf(`/C start "process" "%s"`, execPath)
+	args := []string{}
+
+	if len(extraArgs) != 0 {
+		args = appendArgIfNotPresent(os.Args[1:], extraArgs)
+	} else {
+		args = os.Args[1:]
+	}
+
+	if len(args) > 0 {
+		cmdLine += " " + strings.Join(args, " ")
+	}
+
+	cmd := exec.Command("cmd.exe")
+	cmd.SysProcAttr = &syscall.SysProcAttr{CmdLine: cmdLine}
+
+	if err := cmd.Run(); err != nil {
```

Comment:
Doesn't `cmd.Run()` wait for completion btw? I feel like we're only interested in `cmd.Start()` here